Unified Phoenix Charger implementation

premultiply opened the pull request:
This PR unifies all implementations for chargers based on Phoenix Contact "EV-ETH" controllers series. This includes a complete drop of any special implementations for controllers using new or old or special Wallbe firmware.
It does a complete clean-up and implements advanced autodetection of all features and configurations and due to the work included in here it adds a requirement for sponsoring token.

It **replaces** the templates `wallbe` and `wallbe-meter` by alias. Just use `phoenix-ev-eth` template in the future.

It **removes** the outdated legacy templates `wallbe-pre2019` and `wallbe-pre2019-meter`.
If you own a charger with requirement for these templates please just update the firmware to a recent version if you observe any strange (unknown) behavior!

If you currently use a direct device configuration (not using `type: template`) you may adapt your config to the simplified interface. All custom meter config booleans have to be removed.

You'll find all necessary template based settings at the documentation https://docs.evcc.io/docs/devices/chargers.

premultiply committed:
wip

premultiply committed:
update templates & docs

premultiply committed:
wip

premultiply committed:
wip

premultiply committed:
wip

premultiply committed:
remove obsolete tests

premultiply committed:
wip

premultiply committed:
wip

andig committed:
Fix decorators

premultiply committed:
wip

premultiply committed:
add e3dc brand

premultiply committed:
wip

andig commented on charger/phoenix-ev-eth.go:
```diff
@@ -136,47 +190,94 @@ var _ api.ChargeTimer = (*PhoenixEVEth)(nil)
+		return "", err
+	}
+
+	return bytesAsString(b), nil
```
bytesAsString- string(b)?

andig commented on templates/definition/charger/phoenix-ev-eth.yaml:
```diff
@@ -1,21 +1,28 @@
-    en: The Wallbe must be connected using Ethernet and the DIP switch 10 must be set to 'ON'.
-    de: Die Wallbox muss über ein Netzwerkkabel angebunden sein und im Gerät muss der DIP Schalter 10 auf 'ON' gestellt sein.
-  uri: https://docs.evcc.io/docs/devices/chargers#phoenix-contact
+    en: The charger has to be connected via network cable and the DIP switch 10 in the device must be set to 'ON'. A recent controller firmware is recommended.
```
Brauchen wir egtl. diese "mit Kabel" Kommentare? Wie denn sonst???

premultiply replied:
Nicht wirklich. Das kommt noch aus den Anfangstagen.

premultiply committed:
simplify description

premultiply committed:
cleanup reading and binary decoding

andig commented:
@premultiply was fehlt denn hier noch?

andig commented on charger/phoenix-ev-eth.go:
```diff
@@ -1,97 +1,138 @@
 }
 
+const (
+	wbRegStatus          = 100  // Input
```
das Präfix müsst phx bleiben

premultiply committed:
rename reg consts

andig committed:
Kostal: add Pico hybrid (#8609)

andig committed:
Porsche: switch identity provider (#8546)

naltatis committed:
Use estimator only for vehicles with soc (#8663)

* Estimator only for vehicles with soc

* fix tests

andig committed:
Porsche: keep cookies alive

andig committed:
Twc: wake-up vehicle if enable fails (#8684)

andig committed:
Porsche: simplify emobility handling

naltatis committed:
chore: upgrade npm dependencies (#8664)

andig committed:
Always return asleep error even if vehicle wakeup succeeded so loidpoint retries (#8688)

andig committed:
chore: re-add logging

MarkusGH committed:
Wake-up: add more logging (#8687)

andig committed:
Porsche: add token refresh (#8695)

premultiply committed:
foxess-h3 template: fix energy register (#8706)

andig committed:
Dsmr: fix busy wait with ser2net (#8624)

andig committed:
BMW: fix api (#8713)

GrimmiMeloni committed:
Easee: wait for async current update on resume (#8588)

andig committed:
Tariffs: remove unused units (#8723)

thierolm committed:
Go-E: set 1s default cache duration (#8714)

premultiply committed:
Sungrow: Fix documentation (#8732)

weblate committed:
Translations update from Hosted Weblate (#8566)

* Translated using Weblate (Romanian)

Currently translated at 90.4% (245 of 271 strings)

Co-authored-by: bgrigoriu <b_grigoriu@hotmail.com>
Translate-URL: https://hosted.weblate.org/projects/evcc/evcc/ro/
Translation: evcc/evcc

* Translated using Weblate (Dutch)

Currently translated at 99.6% (270 of 271 strings)

Translated using Weblate (Dutch)

Currently translated at 99.6% (270 of 271 strings)

Co-authored-by: pjv <ezelspinguin@gmail.com>
Translate-URL: https://hosted.weblate.org/projects/evcc/evcc/nl/
Translation: evcc/evcc

* Translated using Weblate (Polish)

Currently translated at 100.0% (271 of 271 strings)

Co-authored-by: RTTTC <94727758+RTTTC@users.noreply.github.com>
Translate-URL: https://hosted.weblate.org/projects/evcc/evcc/pl/
Translation: evcc/evcc

* Translated using Weblate (Croatian)

Currently translated at 100.0% (271 of 271 strings)

Translated using Weblate (Croatian)

Currently translated at 100.0% (271 of 271 strings)

Translated using Weblate (Croatian)

Currently translated at 100.0% (271 of 271 strings)

Co-authored-by: Milo Ivir <mail@milotype.de>
Translate-URL: https://hosted.weblate.org/projects/evcc/evcc/hr/
Translation: evcc/evcc

* Translated using Weblate (French)

Currently translated at 100.0% (271 of 271 strings)

Translated using Weblate (French)

Currently translated at 100.0% (271 of 271 strings)

Co-authored-by: Frederic Mauron <fred@mauron.net>
Translate-URL: https://hosted.weblate.org/projects/evcc/evcc/fr/
Translation: evcc/evcc

* Translated using Weblate (Bulgarian)

Currently translated at 12.9% (35 of 271 strings)

Translated using Weblate (Bulgarian)

Currently translated at 10.3% (28 of 271 strings)

Translated using Weblate (Bulgarian)

Currently translated at 5.9% (16 of 271 strings)

Added translation using Weblate (Bulgarian)

Co-authored-by: Martin Valov <martin.valov@gmail.com>
Translate-URL: https://hosted.weblate.org/projects/evcc/evcc/bg/
Translation: evcc/evcc

* fix toml

---------

Co-authored-by: bgrigoriu <b_grigoriu@hotmail.com>
Co-authored-by: pjv <ezelspinguin@gmail.com>
Co-authored-by: RTTTC <94727758+RTTTC@users.noreply.github.com>
Co-authored-by: Milo Ivir <mail@milotype.de>
Co-authored-by: Frederic Mauron <fred@mauron.net>
Co-authored-by: Martin Valov <martin.valov@gmail.com>
Co-authored-by: premultiply <4681172+premultiply@users.noreply.github.com>

thierolm committed:
Homematic: expose cache parameter (#7863)

andig committed:
Keba: align status with modbus version

andig committed:
Remove dedicated Audi type, replace with eTron (#8547) (BC)

andig committed:
Ocpp: fix setting MeterValueSampleInterval

andig committed:
Hyundai: fix registration id (#8766)

andig committed:
Fix charging not stopped when energy limit reached (#8768)

marval committed:
Add support for Energinet (#8717)

andig committed:
Seat: switch from seatconnect to cupraconnect loging (#8773)

DerAndereAndi committed:
Fix `value` generation in configure process (#8776)

The `plant` should be set at the beginning of the configuration proecss and used to set the machine `CustomID` as evcc does when starting up with a config file.

Fixes https://github.com/evcc-io/evcc/issues/8775

andig committed:
Kia: fix login

andig committed:
Ocpp: increase dispatcher timeout to 1m

andig committed:
Easee: refresh token within access token validity time (#8783)

andig committed:
Bluelink: make Kia/Hyundai compatible

andig committed:
Porsche: fix token refresh

andig committed:
Easee: use ReuseTokenSourceWithExpiry for token refresher

stefanknotzer committed:
Tapo: retry login on error (#8733)

andig committed:
Easee: use oauth validity check to utilize tokens expiryDelta

andig committed:
chore: improve readablity

naltatis committed:
Experimental UI: minimum mincurrent to 1A

andig committed:
chore: fork grid-x/modbus

andig committed:
Porsche: validate token every 5min

andig committed:
Easee: validate token every 5min

naltatis committed:
Sessions: ui refinement, mobile table, breadcrumb, filters (#8793)

andig committed:
chore: log sitePower error

RTTTC committed:
Translated using Weblate (Polish)

Currently translated at 100.0% (272 of 272 strings)

Translated using Weblate (Lithuanian)

Currently translated at 100.0% (272 of 272 strings)

Translated using Weblate (Lithuanian)

Currently translated at 99.6% (271 of 272 strings)

Translated using Weblate (Polish)

Currently translated at 100.0% (271 of 271 strings)

Translated using Weblate (Lithuanian)

Currently translated at 100.0% (271 of 271 strings)

Co-authored-by: RTTTC <94727758+RTTTC@users.noreply.github.com>
Translate-URL: https://hosted.weblate.org/projects/evcc/evcc/lt/
Translate-URL: https://hosted.weblate.org/projects/evcc/evcc/pl/
Translation: evcc/evcc

suhaibaffan committed:
Translated using Weblate (Arabic)

Currently translated at 9.9% (27 of 271 strings)

Added translation using Weblate (Arabic)

Co-authored-by: Suhaib Affan <suhaibaffan91@gmail.com>
Translate-URL: https://hosted.weblate.org/projects/evcc/evcc/ar/
Translation: evcc/evcc

kfussmann committed:
Translated using Weblate (Portuguese)

Currently translated at 100.0% (272 of 272 strings)

Co-authored-by: kfussmann <karlfussmann@gmx.de>
Translate-URL: https://hosted.weblate.org/projects/evcc/evcc/pt/
Translation: evcc/evcc

Commit:
Translated using Weblate (Spanish)

Currently translated at 100.0% (272 of 272 strings)

Co-authored-by: gallegonovato <fran-carro@hotmail.es>
Translate-URL: https://hosted.weblate.org/projects/evcc/evcc/es/
Translation: evcc/evcc

VolkerK62 committed:
docs: add octopusenergy to evcc.dist.yaml (#8864)

* add octopusenergy to evcc.dist.yaml

* fmt

---------

Co-authored-by: Michael Geers <michael@geers.tv>

weblate committed:
Merge remote-tracking branch 'origin/master'

naltatis committed:
chore: fmt toml

naltatis committed:
i18n: add Arabic, Bihari and and Czech translation

naltatis committed:
chore: using correct avg sign ⌀ (#8865)

naltatis committed:
Translated using Weblate (German)

Currently translated at 99.2% (270 of 272 strings)

Co-authored-by: Michael Geers <michael@geers.tv>
Translate-URL: https://hosted.weblate.org/projects/evcc/evcc/de/
Translation: evcc/evcc

naltatis committed:
chore: fix en typos (#8869)

RTTTC committed:
Translated using Weblate (Lithuanian)

Currently translated at 100.0% (272 of 272 strings)

Co-authored-by: RTTTC <94727758+RTTTC@users.noreply.github.com>
Translate-URL: https://hosted.weblate.org/projects/evcc/evcc/lt/
Translation: evcc/evcc

Kefir2105 committed:
Translated using Weblate (Ukrainian)

Currently translated at 100.0% (272 of 272 strings)

Co-authored-by: Dan <jonweblin2205@protonmail.com>
Translate-URL: https://hosted.weblate.org/projects/evcc/evcc/uk/
Translation: evcc/evcc

comradekingu committed:
Translated using Weblate (Norwegian Bokmål)

Currently translated at 97.7% (266 of 272 strings)

Co-authored-by: Allan Nordhøy <epost@anotheragency.no>
Translate-URL: https://hosted.weblate.org/projects/evcc/evcc/nb_NO/
Translation: evcc/evcc

VolkerK62 committed:
Fix minActiveVoltage (#8867)

Reduce lower limit to >= 207V (230V-10%)

weblate committed:
Merge remote-tracking branch 'origin/master'

JanDragon committed:
chore: update deprecated GoReleaser tap use (#8859)

In the brews section of GoReleaser, an update has been made regarding the utilization of the "tap" functionality. As of June 13, 2023, "tap" has been marked as deprecated. It is important to note that this change does not impact the functionality or behavior of the release mechanism.

weblate committed:
Merge remote-tracking branch 'origin/master'

naltatis committed:
chore: toml

andig committed:
chore: reduce logging (#8880)

andig committed:
NRG: upgrade muka/go-bluetooth to 5.64 for Debian 12

andig committed:
chore: remove unused tokensource property (#8888)

naltatis committed:
chore: upgrade npm dependencies (#8882)

* Chore: npm dependency upgrades

* eslint <> prettier 3 compat

* devDeps > deps

tuetenk0pp committed:
Add Inverter Steca coolcept fleX (#8861)

Duell10111 committed:
Keba: fix vehicle connected status when only the cable is connected to the charging station (#8898)

andig committed:
Fix taking vehicle priorities into account (#8884)

andig committed:
Easee: reduce logging (#8881)

andig committed:
Seat: refer to official api generations (#8774)

premultiply committed:
Add Weidmüller AC Smart (#7340)

weblate committed:
Translated using Weblate (Spanish) (#8886)

Currently translated at 100.0% (272 of 272 strings)


Translate-URL: https://hosted.weblate.org/projects/evcc/evcc/es/
Translation: evcc/evcc

Co-authored-by: gallegonovato <fran-carro@hotmail.es>

dependabot[bot] committed:
Core: Bump semver from 5.7.1 to 5.7.2 (#8906)

Bumps [semver](https://github.com/npm/node-semver) from 5.7.1 to 5.7.2.
- [Release notes](https://github.com/npm/node-semver/releases)
- [Changelog](https://github.com/npm/node-semver/blob/v5.7.2/CHANGELOG.md)
- [Commits](https://github.com/npm/node-semver/compare/v5.7.1...v5.7.2)

---
updated-dependencies:
- dependency-name: semver
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

premultiply committed:
Add M-TEC Energy Butler (#8901)

ngehrsitz committed:
Add Sonnenbatterie comfort, eco 5, eco 6, oem 6.5 (#8679)

andig committed:
Easee: prevent DOS when signalr connect fails due to rejected token (#8915)

andig committed:
Tibber: add configurable charges and taxes (#8923)

andig committed:
Tariffs: remove deprecated currency config (BC) (#8924)

weblate committed:
Translations update from Hosted Weblate (#8903)

* Translated using Weblate (Spanish)

Currently translated at 100.0% (272 of 272 strings)

Co-authored-by: gallegonovato <fran-carro@hotmail.es>
Translate-URL: https://hosted.weblate.org/projects/evcc/evcc/es/
Translation: evcc/evcc

* Translated using Weblate (Swedish)

Currently translated at 85.2% (232 of 272 strings)

Translation: evcc/evcc
Translate-URL: https://hosted.weblate.org/projects/evcc/evcc/sv/

---------

Co-authored-by: gallegonovato <fran-carro@hotmail.es>
Co-authored-by: Michael Geers <michael@geers.tv>

StefanSchoof committed:
Add `npm install` to Contribute in README (#8936)

* Add `npm install`

See #8920

* Update README.md

Co-authored-by: Michael Geers <michael@geers.tv>

---------

Co-authored-by: Michael Geers <michael@geers.tv>

premultiply committed:
Add Youless Energy Meter (#8941)

* power
* energy total import

andig committed:
Porsche: remove secondary identity

naltatis committed:
i18n: energy format in subunit (#8937)

* i18n: energy format in subunit

* test

premultiply committed:
Device-specific OCPP template improvements (#8900)

EVBOX Elvi and Wallbox Pulsar Plus / Commander

naltatis committed:
docs: markdown compatible placeholder

andig committed:
Plugins: trace xml transformations (#8879)

andig committed:
chore: warn on intervals below 30s

andig committed:
chore: set 30s default interval

andig committed:
Porsche: refresh tokens within 1h (#8990)

premultiply committed:
Fix M-tec templates (#8993)

andig committed:
Volvo Connected: fix token refresh (#8998)

andig committed:
Tibber: use energy instead of total as base if charges or taxes are defined (#9000)

andig committed:
VAG: fix login (#9009)

andig committed:
chore: refactor forced token refresh

andig committed:
Mqtt: fix setter not cleared

salz3n committed:
Enphase: Add battery (#9019)

Co-authored-by: Sjard Bleecken <sjard.bleecken@fitreisen.de>

premultiply committed:
Youless: Add pv meter (#9023)

weblate committed:
Translations update from Hosted Weblate (#8935)

* Translated using Weblate (Spanish)

Currently translated at 100.0% (272 of 272 strings)

Co-authored-by: gallegonovato <fran-carro@hotmail.es>
Translate-URL: https://hosted.weblate.org/projects/evcc/evcc/es/
Translation: evcc/evcc

* Translated using Weblate (Swedish)

Currently translated at 85.2% (232 of 272 strings)

Translation: evcc/evcc
Translate-URL: https://hosted.weblate.org/projects/evcc/evcc/sv/

* Translated using Weblate (Danish)

Currently translated at 100.0% (272 of 272 strings)

Translation: evcc/evcc
Translate-URL: https://hosted.weblate.org/projects/evcc/evcc/da/

* fix toml

---------

Co-authored-by: gallegonovato <fran-carro@hotmail.es>
Co-authored-by: Michael Geers <michael@geers.tv>
Co-authored-by: Philipp <ph.bohk@gmail.com>
Co-authored-by: premultiply <4681172+premultiply@users.noreply.github.com>

andig committed:
Niu: fix api (#9040)

sparkyson committed:
OpenEVSE: fix status handling (#9015)

premultiply committed:
Publish total energies (#8905)

premultiply committed:
Add ABL eM4 (#9052)

premultiply committed:
site: fix individual pv+bat energy

thierolm committed:
Tapo: add missing login errorcheck (#9056)

GrimmiMeloni committed:
Easee: ignore outdated observations (#9053)

DerAndereAndi committed:
Elli: clarify which wallboxes are supported best (#9036)

The current Elli wallboxes contain lots of software bugs, which make their usage and also support a nightmare.

Only the following two scenarios provide a somehow stable environment which can be “recommended”:

- Elli Pro: No CT coil usage, LAN connection. If a LAN connection is not possible, addition of an external meter
- Elli Pro: No CT coil usage, external meter usage.

A LAN connection is always highly recommended, as even the slightest network connection drop will allow charging of the car with minimal power until reconnection is establied.

There is no workaround for the wallbox software crashing often on connection attempts which can take a few minutes for the connection to work. During that time the EV is allowed to charge with minimal power.

GrimmiMeloni committed:
Easee: do not wait for DCC update after NoOp API call (#9018)

andig committed:
chore: refactor retryMeter

andig committed:
BMW: refactor token generation

andig committed:
chore: align naming

andig committed:
chore: use integrated pv system as term (#9076)

premultiply committed:
Add Slimmelezer ESPHome DSMR (#9087)

premultiply committed:
M-Tec: fix id default and grid power scale (#9084)

andig committed:
Porsche: fix nil panic after failed token refresh

andig committed:
Volvo Connected: fix token refresh

thierolm committed:
Add HomeWizard (#9102)

andig committed:
ABL: clarify eMH versions (#9077)

andig committed:
cmd/vehicle: error if vehicle creation has failed (#9130)

StefanSchoof committed:
chore: fix typo (#9131)

andig committed:
Tariffs: retry download with backoff (#8833)

andig committed:
chore: fix token refresh

andig committed:
Add CurrentGetter api for validating charger current (#9147)

VolkerK62 committed:
Add Jeep (#9140)

andig committed:
Easee: implement api.CurrentGetter (#9148)

andig committed:
Don't force-sync charger prematurely (#9075)

andig committed:
Bluelink: calculate stamps locally (#9170)

andig committed:
Homematic: refactor (#9172)

thierolm committed:
Homewizard: add cache (#9158)

GrimmiMeloni committed:
Easee: wait for op_mode change when pausing/resuming (#9156)

premultiply committed:
API: extend charger status mapping (#9142)

weblate committed:
Translations update from Hosted Weblate (#9070)

* Translated using Weblate (Croatian)

Currently translated at 100.0% (272 of 272 strings)

Co-authored-by: Milo Ivir <mail@milotype.de>
Translate-URL: https://hosted.weblate.org/projects/evcc/evcc/hr/
Translation: evcc/evcc

* Translated using Weblate (German)

Currently translated at 100.0% (272 of 272 strings)

Co-authored-by: dblndr <c@dablander.org>
Translate-URL: https://hosted.weblate.org/projects/evcc/evcc/de/
Translation: evcc/evcc

* fix toml

---------

Co-authored-by: Milo Ivir <mail@milotype.de>
Co-authored-by: dblndr <c@dablander.org>
Co-authored-by: premultiply <4681172+premultiply@users.noreply.github.com>

andig committed:
Use first of month for prev/next calculation (#9197)

naltatis committed:
Energyflow: kW/W switch for non-changing values (#9206)

* Energyflow: fix kW/W switch for non-changing values

* Energyflow: fix kW/W switch for non-changing values

* fix test

naltatis committed:
Persist targetSoc/Energy, minSoc, targetTime per vehicle, add minSoc to UI (#8115)

* Plugins: make javascript return values more permissive

* Persist targetSoc/Energy, minSoc and target time, remove experimental from minSoc UI

* go generate

* Fix tests

* Update minSoc description

* Store settings by index. Show plan outside pv mode. Add help texts.

* remove slug

* go mock

* lint

* chore: simplify templates (#8304)

* Add e2e tests with playwright (#8123)

* Revert "1p3p: let charger handle session stop/restart (#7723)"

This reverts commit dd787ce52fe07e617a73061fbb31393335f381dd.

* mazda2mqtt: document vin required (#8319)

* FoxESS: split H1/H3 devices (#7376)

Co-authored-by: premultiply <4681172+premultiply@users.noreply.github.com>

* Translations update from Hosted Weblate (#8124)

* Translated using Weblate (Czech)

Currently translated at 37.3% (99 of 265 strings)

Co-authored-by: Dusan Suja <bc.suja.dusan@googlemail.com>
Translate-URL: https://hosted.weblate.org/projects/evcc/evcc/cs/
Translation: evcc/evcc

* Translated using Weblate (Finnish)

Currently translated at 100.0% (265 of 265 strings)

Co-authored-by: Arna Lepikkö <arna.lepikko@telemail.fi>
Translate-URL: https://hosted.weblate.org/projects/evcc/evcc/fi/
Translation: evcc/evcc

* Translated using Weblate (German)

Currently translated at 100.0% (265 of 265 strings)

Co-authored-by: ThinkEV <claas@rootdir.de>
Translate-URL: https://hosted.weblate.org/projects/evcc/evcc/de/
Translation: evcc/evcc

* Translated using Weblate (Croatian)

Currently translated at 100.0% (265 of 265 strings)

Translated using Weblate (Slovenian)

Currently translated at 100.0% (265 of 265 strings)

Co-authored-by: Žiga Deisinger <ziga@deisinger.si>
Translate-URL: https://hosted.weblate.org/projects/evcc/evcc/hr/
Translate-URL: https://hosted.weblate.org/projects/evcc/evcc/sl/
Translation: evcc/evcc

* Translated using Weblate (Catalan)

Currently translated at 100.0% (265 of 265 strings)

Co-authored-by: Norbert Poch <npoch@yahoo.com>
Translate-URL: https://hosted.weblate.org/projects/evcc/evcc/ca/
Translation: evcc/evcc

* Translated using Weblate (Dutch)

Currently translated at 76.6% (203 of 265 strings)

Translated using Weblate (Dutch)

Currently translated at 74.3% (197 of 265 strings)

Co-authored-by: Ruben Van Boxem <vanboxem.ruben@gmail.com>
Translate-URL: https://hosted.weblate.org/projects/evcc/evcc/nl/
Translation: evcc/evcc

* fix toml

---------

Co-authored-by: Dusan Suja <bc.suja.dusan@googlemail.com>
Co-authored-by: Arna Lepikkö <arna.lepikko@telemail.fi>
Co-authored-by: ThinkEV <claas@rootdir.de>
Co-authored-by: Žiga Deisinger <ziga@deisinger.si>
Co-authored-by: Norbert Poch <npoch@yahoo.com>
Co-authored-by: Ruben Van Boxem <vanboxem.ruben@gmail.com>
Co-authored-by: premultiply <4681172+premultiply@users.noreply.github.com>

* Add OBO Betterman Ion (#8321)

Also aligns heartbeat implementations

* Enphase: add token auth (firmware D7.x.xxx) and grid (#8247)

* Mqtt: fix smartCostLimit topic case (#8328)

* Add ISO15118 vehicle template (#8302)

* Mqtt: simplify setters

* Mqtt: disable message ordering to improve performance

* chore: simplify random state generation

* Porsche: remove deprecated mobile api (#8349)

* Porsche: remove remaining mobile api types

* Cupra: add odometer (#8340)

* chore: refactor go-stylish

* chore: fix stale handler

* mazda2mqtt: longer timeout (#8364)

* chore: pre-process toml

* chore: prevent toml double quotes

* Audi: temporarily switch to etron (#8374)

* Fix nightly build (#8384)

* Update SunSpec templates (#8270)

* Revert "Fix nightly build (#8384)"

This reverts commit 536dbc936c7ced16f8bd00409a8cc00e5e807e57.

* Revert "Add e2e tests with playwright (#8123)"

This reverts commit 0df815729b5471bc7c448857d60137684b926611.

* Easee: update Observation IDs (#8391)

* Easee: handle smartCharging errors (#8389)

* TWC: add non-Tesla vehicle warning (#8329)

* TWC: allow loadpoint to wakeup vehicle (#8284)

* chore: use gridx consts

* Fix ISO15118 vehicle (#8402)

Such vehicles would be reported as `(Offline)` previously, as no Soc can be fetched and thus needs to behave as the `offline` template.

The SoC is fetched via the charger, so hardcode it to 0 in here

* Modbus: add coils  (#8385)

* OpenEVSE: fix api (#8415)

* chore: refactor offline vehicles (#8404)

* Easee: wait for api confirmation (#8307)

* Revert "Easee: wait for api confirmation (#8307)"

This reverts commit ce0e4277aa9b9f7b58c12d7bf0958ae7f632c729.

* automatically switch session log energy unit (#8371)

Show the charged energy in the session log table and details in Wh for
values below 1 kWh and in kWh for larger values

* Fix nightly/release + integration workflow  (#8450)

* Revert "Add e2e tests with playwright (#8123)"

This reverts commit 0df815729b5471bc7c448857d60137684b926611.

* Revert "Revert "Add e2e tests with playwright (#8123)""

This reverts commit 2299306954045dfe62541d5520a66b5ab7ec4971.

* fix nightly/release

* fix nightly/release

* fix nightly/release

---------

Co-authored-by: andig <cpuidle@gmx.de>

* chore: no integration on nightly

* chore: no integration on nightly

* chore: no integration on nightly

* Sessions: filter, monthly sums, paging (#8278)

* fmt

* Update i18n/en.toml

Co-authored-by: Simon Riepl <43091717+savus4@users.noreply.github.com>

* Use exp/slices

* Use mux consistently

* add integration tests

---------

Co-authored-by: andig <cpuidle@gmx.de>
Co-authored-by: C64Axel <33828760+C64Axel@users.noreply.github.com>
Co-authored-by: premultiply <4681172+premultiply@users.noreply.github.com>
Co-authored-by: Weblate (bot) <hosted@weblate.org>
Co-authored-by: Dusan Suja <bc.suja.dusan@googlemail.com>
Co-authored-by: Arna Lepikkö <arna.lepikko@telemail.fi>
Co-authored-by: ThinkEV <claas@rootdir.de>
Co-authored-by: Žiga Deisinger <ziga@deisinger.si>
Co-authored-by: Norbert Poch <npoch@yahoo.com>
Co-authored-by: Ruben Van Boxem <vanboxem.ruben@gmail.com>
Co-authored-by: salz3n <79696598+salz3n@users.noreply.github.com>
Co-authored-by: Daniel Paschke <paschdan@gmail.com>
Co-authored-by: Andreas Linde <42185+DerAndereAndi@users.noreply.github.com>
Co-authored-by: lex777777 <84906358+lex777777@users.noreply.github.com>
Co-authored-by: Oscar <OAltr@users.noreply.github.com>
Co-authored-by: Michael Heß <GrimmiMeloni@users.noreply.github.com>
Co-authored-by: RTTTC <94727758+RTTTC@users.noreply.github.com>
Co-authored-by: kscholty <47229207+kscholty@users.noreply.github.com>
Co-authored-by: Jan Alexander <jan.alexander@posteo.de>
Co-authored-by: Simon Riepl <43091717+savus4@users.noreply.github.com>

andig committed:
Discovergy: add timeout (#9212)

naltatis committed:
fix session detail currency formatting (#9230)

andig committed:
RCT: add retry (#9231)

andig committed:
Always publish status

andig committed:
Optimistically publish status after updating

andig committed:
chore: fix double-publish

andig committed:
ABL: add timeout (#9241)

andig committed:
OpenWB-Pro: support reading initial soc (#9247)

andig committed:
chore: trim local build information (#9246)

naltatis committed:
UI: Support negative prices and smart cost limit (#9233)

* UI: negative smart cost limit

* use en dash

weblate committed:
Translations update from Hosted Weblate (#9213)

* Translated using Weblate (Polish)

Currently translated at 100.0% (274 of 274 strings)

Translated using Weblate (Lithuanian)

Currently translated at 100.0% (274 of 274 strings)

Co-authored-by: RTTTC <94727758+RTTTC@users.noreply.github.com>
Translate-URL: https://hosted.weblate.org/projects/evcc/evcc/lt/
Translate-URL: https://hosted.weblate.org/projects/evcc/evcc/pl/
Translation: evcc/evcc

* Translated using Weblate (Portuguese)

Currently translated at 100.0% (274 of 274 strings)

Co-authored-by: kfussmann <karlfussmann@gmx.de>
Translate-URL: https://hosted.weblate.org/projects/evcc/evcc/pt/
Translation: evcc/evcc

* Translated using Weblate (Croatian)

Currently translated at 100.0% (274 of 274 strings)

Co-authored-by: Milo Ivir <mail@milotype.de>
Translate-URL: https://hosted.weblate.org/projects/evcc/evcc/hr/
Translation: evcc/evcc

* Translated using Weblate (Czech)

Currently translated at 46.3% (127 of 274 strings)

Co-authored-by: Dusan Suja <bc.suja.dusan@googlemail.com>
Translate-URL: https://hosted.weblate.org/projects/evcc/evcc/cs/
Translation: evcc/evcc

* Translated using Weblate (Spanish)

Currently translated at 100.0% (274 of 274 strings)

Co-authored-by: gallegonovato <fran-carro@hotmail.es>
Translate-URL: https://hosted.weblate.org/projects/evcc/evcc/es/
Translation: evcc/evcc

* Translated using Weblate (Finnish)

Currently translated at 100.0% (274 of 274 strings)

Co-authored-by: Arna Lepikkö <arna.lepikko@telemail.fi>
Translate-URL: https://hosted.weblate.org/projects/evcc/evcc/fi/
Translation: evcc/evcc

* Translated using Weblate (French)

Currently translated at 98.5% (270 of 274 strings)

Co-authored-by: kjorand <kevin.jorand@hotmail.com>
Translate-URL: https://hosted.weblate.org/projects/evcc/evcc/fr/
Translation: evcc/evcc

* lint

---------

Co-authored-by: RTTTC <94727758+RTTTC@users.noreply.github.com>
Co-authored-by: kfussmann <karlfussmann@gmx.de>
Co-authored-by: Milo Ivir <mail@milotype.de>
Co-authored-by: Dusan Suja <bc.suja.dusan@googlemail.com>
Co-authored-by: gallegonovato <fran-carro@hotmail.es>
Co-authored-by: Arna Lepikkö <arna.lepikko@telemail.fi>
Co-authored-by: kjorand <kevin.jorand@hotmail.com>
Co-authored-by: Michael Geers <michael@geers.tv>

andig committed:
Versicharge: update register map (#9287)

panzerdev committed:
VW ID: implement api.Resurrector to start charging from sleep (#9276)

andig committed:
chore: minor

naltatis committed:
Settings/Savings: persist unsaved values regularly (#9257)

andig committed:
chore: improve trimming string from bytes

andig committed:
chore: upgrade and fork go-powerwall

andig committed:
chore: fix forked go-powerwall

andig committed:
chore: lazy-start heartbeat

andig committed:
Fix potential busy loop with short timeouts (#9320)

andig committed:
Heidelberg EC: handle failsafe timeout (#9322)

andig committed:
chore: fix conversion

poohnet committed:
Fix & simplify SMA Data Manager template (#9323)

andig committed:
Plugins: let charger support api.ChargerEx (#9331)

andig committed:
chore: use charge status mapper (#9305)

premultiply committed:
Docs: add missing 1p3p capabilities (#9334)

naltatis committed:
Fix odometer number format (#9345)

premultiply committed:
Enphase: fix soc and power (#9332)

* fix soc and power readings

* remove unspecified grid import energy

* remove cache

* wip

premultiply committed:
Add Acrel ADW300 (#9330)

VolkerK62 committed:
Shelly: set meter default channel 0 (#9362)

DerAndereAndi committed:
Improve EEBUS isCharging detection (#9354)

- Make sure the check using an external meter is identical to using an internal meter
- Fix check using internal meter to compare sum of all charge currents to idlefactor * min. single phase charge current

andig committed:
Ensure currents get synced (#9369)

andig committed:
Warp2: refactor ChargerEx implementation

andig committed:
Make csv export more flexible (#9309)

andig committed:
chore: fix min current register

naltatis committed:
Allow set float for http plugin (#9381)

andig committed:
Custom charger: decorate soc (#9382)

premultiply committed:
SMA Data Manager: enhance template (#9329)

naltatis committed:
Heating: basic support (#9393)

andig committed:
Mqtt: update online status on reconnect (#9394)

carygravel committed:
i18n: tidy up English translation of config cmd (#9356)

* i18n: tidy up English translation of config cmd

* revert loadpoint -> installation

naltatis committed:
Sessions: add charge duration (#9263)

weblate committed:
Translations update from Hosted Weblate (#9365)

* Translated using Weblate (Ukrainian)

Currently translated at 100.0% (274 of 274 strings)

Co-authored-by: Dan <jonweblin2205@protonmail.com>
Translate-URL: https://hosted.weblate.org/projects/evcc/evcc/uk/
Translation: evcc/evcc

* Translated using Weblate (Portuguese)

Currently translated at 100.0% (278 of 278 strings)

Translated using Weblate (Portuguese)

Currently translated at 100.0% (276 of 276 strings)

Co-authored-by: kfussmann <karlfussmann@gmx.de>
Translate-URL: https://hosted.weblate.org/projects/evcc/evcc/pt/
Translation: evcc/evcc

* Translated using Weblate (Polish)

Currently translated at 100.0% (278 of 278 strings)

Translated using Weblate (Lithuanian)

Currently translated at 100.0% (278 of 278 strings)

Translated using Weblate (Lithuanian)

Currently translated at 100.0% (276 of 276 strings)

Co-authored-by: RTTTC <94727758+RTTTC@users.noreply.github.com>
Translate-URL: https://hosted.weblate.org/projects/evcc/evcc/lt/
Translate-URL: https://hosted.weblate.org/projects/evcc/evcc/pl/
Translation: evcc/evcc

* Translated using Weblate (Spanish)

Currently translated at 100.0% (278 of 278 strings)

Co-authored-by: gallegonovato <fran-carro@hotmail.es>
Translate-URL: https://hosted.weblate.org/projects/evcc/evcc/es/
Translation: evcc/evcc

* toml

---------

Co-authored-by: Dan <jonweblin2205@protonmail.com>
Co-authored-by: kfussmann <karlfussmann@gmx.de>
Co-authored-by: RTTTC <94727758+RTTTC@users.noreply.github.com>
Co-authored-by: gallegonovato <fran-carro@hotmail.es>
Co-authored-by: Michael Geers <michael@geers.tv>

weblate committed:
Translations update from Hosted Weblate (#9405)

* Translated using Weblate (Ukrainian)

Currently translated at 100.0% (274 of 274 strings)

Co-authored-by: Dan <jonweblin2205@protonmail.com>
Translate-URL: https://hosted.weblate.org/projects/evcc/evcc/uk/
Translation: evcc/evcc

* Translated using Weblate (Portuguese)

Currently translated at 100.0% (278 of 278 strings)

Translated using Weblate (Portuguese)

Currently translated at 100.0% (276 of 276 strings)

Co-authored-by: kfussmann <karlfussmann@gmx.de>
Translate-URL: https://hosted.weblate.org/projects/evcc/evcc/pt/
Translation: evcc/evcc

* Translated using Weblate (Polish)

Currently translated at 100.0% (278 of 278 strings)

Translated using Weblate (Lithuanian)

Currently translated at 100.0% (278 of 278 strings)

Translated using Weblate (Lithuanian)

Currently translated at 100.0% (276 of 276 strings)

Co-authored-by: RTTTC <94727758+RTTTC@users.noreply.github.com>
Translate-URL: https://hosted.weblate.org/projects/evcc/evcc/lt/
Translate-URL: https://hosted.weblate.org/projects/evcc/evcc/pl/
Translation: evcc/evcc

* Translated using Weblate (Spanish)

Currently translated at 100.0% (278 of 278 strings)

Co-authored-by: gallegonovato <fran-carro@hotmail.es>
Translate-URL: https://hosted.weblate.org/projects/evcc/evcc/es/
Translation: evcc/evcc

* toml

---------

Co-authored-by: Dan <jonweblin2205@protonmail.com>
Co-authored-by: kfussmann <karlfussmann@gmx.de>
Co-authored-by: RTTTC <94727758+RTTTC@users.noreply.github.com>
Co-authored-by: gallegonovato <fran-carro@hotmail.es>
Co-authored-by: Michael Geers <michael@geers.tv>

andig committed:
Sessions: fix formatting of pointer values (#9406)

GrimmiMeloni committed:
Loadpoint: follow charger current (#9408)

JanDragon committed:
chore: allow concurrent use of Tee (#9396)

GrimmiMeloni committed:
Easee: improve enabled detection (#9314)

naltatis committed:
Fix resetOnDisconnect overrides vehicle settings (BC) (#9370)

andig committed:
Add support for loading and storing device configuration from/to database (#6199)

andig committed:
Require Go 1.21 (#9244)

naltatis committed:
config: fix icon select test

andig committed:
chore: sort configurable vehicles by id

andig committed:
chore: reorder vehicle params for config ui (#9420)

thierolm committed:
Tasmota: add cache (#9422)

andig committed:
Mqtt: fix panic

andig committed:
chore: fix configuration id deleted on update

snowball77 committed:
Fix SE Home Charger broken by 1p3p (#9440)

andig committed:
chore: add fallback certificates in case os doesn't have any

andig committed:
MyStrom: fix charger template

andig committed:
Add Schneider EVlink (#9436)

weblate committed:
Translations update from Hosted Weblate (#9410)

* Translated using Weblate (French)

Currently translated at 97.4% (271 of 278 strings)

Co-authored-by: kjorand <kevin.jorand@hotmail.com>
Translate-URL: https://hosted.weblate.org/projects/evcc/evcc/fr/
Translation: evcc/evcc

* Translated using Weblate (Lithuanian)

Currently translated at 100.0% (298 of 298 strings)

Co-authored-by: RTTTC <94727758+RTTTC@users.noreply.github.com>
Translate-URL: https://hosted.weblate.org/projects/evcc/evcc/lt/
Translation: evcc/evcc

* Translated using Weblate (Norwegian Bokmål)

Currently translated at 94.2% (281 of 298 strings)

Co-authored-by: Allan Nordhøy <epost@anotheragency.no>
Translate-URL: https://hosted.weblate.org/projects/evcc/evcc/nb_NO/
Translation: evcc/evcc

* Translated using Weblate (Portuguese)

Currently translated at 100.0% (298 of 298 strings)

Co-authored-by: kfussmann <karlfussmann@gmx.de>
Translate-URL: https://hosted.weblate.org/projects/evcc/evcc/pt/
Translation: evcc/evcc

* Translated using Weblate (Spanish)

Currently translated at 100.0% (298 of 298 strings)

Co-authored-by: gallegonovato <fran-carro@hotmail.es>
Translate-URL: https://hosted.weblate.org/projects/evcc/evcc/es/
Translation: evcc/evcc

* fix toml

---------

Co-authored-by: kjorand <kevin.jorand@hotmail.com>
Co-authored-by: RTTTC <94727758+RTTTC@users.noreply.github.com>
Co-authored-by: Allan Nordhøy <epost@anotheragency.no>
Co-authored-by: kfussmann <karlfussmann@gmx.de>
Co-authored-by: gallegonovato <fran-carro@hotmail.es>
Co-authored-by: premultiply <4681172+premultiply@users.noreply.github.com>

premultiply committed:
use encoding

premultiply committed:
use encoding

andig committed:
Add struct validation

premultiply committed:
Schneider: Fix enable/disable (#9459)

GrimmiMeloni committed:
Elapse guard when loadpoint is out of sync (#9450)

andig committed:
Tesla: ignore is_charging error

Hofyyy commented:
Sehe ich das richtig, dass die Änderungen nicht abwärtskompatibel sind und jeder "Nutzer" an die Config ran muss?

naltatis committed:
chore: optimize bundle size (#9464)

andig committed:
Fix reducing to 1p when vehicle doesn't utilize allowed current (#9462)

premultiply committed:
Schneider: fix ChargedEnergy and init (#9461)

premultiply committed:
Enphase: revert API change (#9479)

premultiply committed:
Heidelberg: add Amperfied chargers (#9485)

andig committed:
chore: fix potential copy issue

ElinaUrbanovich committed:
Renault: add position and start/stop charge (#9499)

Co-authored-by: Elina Urbanovich <e.urbanovich@andersenlab.com>

StefanSchoof committed:
Docs: use English weekdays in evcc.dist.yaml (#9509)

andig committed:
Fix panic when charge status is empty string

premultiply committed:
Enphase: fix scale and use HTTPS

premultiply committed:
Amperfied/Heidelberg: split templates (#9505)

andig committed:
Fix resuming charging after phase switch (#9538)

andig committed:
Guard soc against invalid values (#9532)

premultiply committed:
evsewifi: fix session time and meter detection (#9484)

premultiply committed:
Enphase: use HTTP only

weblate committed:
Translations update from Hosted Weblate (#9518)

* Translated using Weblate (Ukrainian)

Currently translated at 100.0% (298 of 298 strings)

Translated using Weblate (Ukrainian)

Currently translated at 97.9% (292 of 298 strings)

Co-authored-by: Dan <jonweblin2205@protonmail.com>
Translate-URL: https://hosted.weblate.org/projects/evcc/evcc/uk/
Translation: evcc/evcc

* Translated using Weblate (Slovenian)

Currently translated at 100.0% (298 of 298 strings)

Translated using Weblate (Slovenian)

Currently translated at 91.6% (273 of 298 strings)

Co-authored-by: Žiga Deisinger <ziga@deisinger.si>
Translate-URL: https://hosted.weblate.org/projects/evcc/evcc/sl/
Translation: evcc/evcc

* fix toml

---------

Co-authored-by: Dan <jonweblin2205@protonmail.com>
Co-authored-by: Žiga Deisinger <ziga@deisinger.si>
Co-authored-by: premultiply <4681172+premultiply@users.noreply.github.com>

MarkusGH committed:
Porsche: disable forced token refresh (#9554)

naltatis committed:
Docs: localize documentation templates (#9508)

andig committed:
Revert "Fix reducing to 1p when vehicle doesn't utilize allowed current (#9462)"

This reverts commit fc963abb79ece522c1440e495bbf64e562ddd0cf.

DerAndereAndi committed:
Fix EEBUS enable detection (#9574)

- If the EV is charging, make sure to report as being enabled
- Fixes https://github.com/evcc-io/evcc/issues/9473

naltatis committed:
chore: cleanup generated templates

premultiply committed:
Senec: add https and energy (#9492)

GrimmiMeloni committed:
Easee: use configured timeout for async waits (#9568)

GrimmiMeloni committed:
Always treat charger as enabled when charging  (#9570)

premultiply committed:
Senec: drop energy (#9608)

premultiply committed:
Senec: make HTTPS default and remove orphan docs (#9612)

andig committed:
Planner: improve logging

andig committed:
Quickly recover from logic error by expiring pv timer (#9624)

premultiply committed:
remove ChargingTime()

premultiply committed:
fix docs and brands

premultiply committed:
Merge branch 'fix/wallbe' of https://github.com/evcc-io/evcc into fix/wallbe

premultiply committed:
clean docs

andig commented:
Der PR hat jetzt 644 veränderter Dateien. Ich empfehle `git rebase`.

premultiply commented:
Ups...
